src/overlays/actors/ovl_En_Kz/z_en_kz.c

```diff
@@ -21,47 +28,458 @@ const ActorInit En_Kz_InitVars = {
+    Actor_SetScale(&this->actor, 0.01);
+    this->actor.unk_1F = 3;
+    this->unk_1E0.unk_00 = 0;
+    func_80034EC0(&this->skelanime, sAnimations, 0);
+
+    if (gSaveContext.eventChkInf[3] & 8) {
+        EnKz_SetMovedPos(this, globalCtx);
+    }
+
+    if (LINK_IS_ADULT) {
+        if (!(gSaveContext.infTable[19] & 0x100)) {
+            Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_BG_ICE_SHELTER,
+                                this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0,
+                                0x04FF);
+        }
+        this->actionFunc = EnKz_Wait;
+    } else {
+        this->actionFunc = EnKz_PreMweepWait;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9C658.s")
+void EnKz_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKz* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9C6C0.s")
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9C8E4.s")
+void EnKz_PreMweepWait(EnKz* this, GlobalContext* globalCtx) {
+    if (this->unk_1E0.unk_00 == 2) {
+        func_80034EC0(&this->skelanime, sAnimations, 2);
+        this->unk_1E0.unk_00 = 0;
+        this->actionFunc = EnKz_SetupMweep;
+    } else {
+        func_80034F54(globalCtx, &this->unk_2A6, &this->unk_2BE, 12);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9C95C.s")
+void EnKz_SetupMweep(EnKz* this, GlobalContext* globalCtx) {
+    Vec3f unused = { 0.0f, 0.0f, 0.0f };
+    Vec3f pos;
+    Vec3f initPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9CB18.s")
+    this->cutsceneCamera = Gameplay_CreateSubCamera(globalCtx);
+    this->gameplayCamera = globalCtx->activeCamera;
+    Gameplay_ChangeCameraStatus(globalCtx, this->gameplayCamera, 1);
+    Gameplay_ChangeCameraStatus(globalCtx, this->cutsceneCamera, 7);
+    pos = this->actor.posRot.pos;
+    initPos = this->actor.initPosRot.pos;
+    pos.y += 60.0f;
+    initPos.y += -100.0f;
+    initPos.z += 260.0f;
+    func_800C04D8(globalCtx, this->cutsceneCamera, &pos, &initPos);
+    func_8002DF54(globalCtx, &this->actor, 8);
+    this->actor.speedXZ = 0.1f;
+    this->actionFunc = EnKz_Mweep;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9CCD8.s")
+void EnKz_Mweep(EnKz* this, GlobalContext* globalCtx) {
+    Vec3f unused = { 0.0f, 0.0f, 0.0f };
+    Vec3f pos;
+    Vec3f initPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9CE44.s")
+    pos = this->actor.posRot.pos;
+    initPos = this->actor.initPosRot.pos;
+    pos.y += 60.0f;
+    initPos.y += -100.0f;
+    initPos.z += 260.0f;
+    func_800C04D8(globalCtx, this->cutsceneCamera, &pos, &initPos);
+    if (EnKz_FollowPath(this, globalCtx) == 1 && (this->waypoint == 0)) {
```

Comment:
I think we want this condition to have parens around it like `(EnKz_FollowPath(this, globalCtx) == 1)`?